luigi/task.py

```diff
@@ -636,6 +639,6 @@ def flatten_output(task):
     """
     r = flatten(task.output())
     if not r:
-        for dep in flatten(task.requires()):
+        for dep in flatten(task.deps()):
```

Comment:
Why would you `flatten` an already `flatten`ed struct? Bc this would be something like `flatten(tasks.(flatten(self._requires()))` (incorrect syntax for the purpose of getting my question across).